### PR TITLE
[duckdb-pyspark] DuckDBPySparkIOManager

### PR DESCRIPTION
**Pull request by jamiedemaria**
## Summary & Motivation
converts the pyspark type handler and io manager to the new resource system

## How I Tested These Changes
unit tests now test both versions of the io manager

### CODE REVIEW

**Comment by jamiedemaria** [comment]
Current dependencies on/for this PR:
* master
  * **PR #12695** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/12695" target="_blank">Graphite</a> 
    * **PR #13221** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/13221" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/13221?utm_source=stack-comment).

File: python_modules/libraries/dagster-duckdb-pyspark/dagster_duckdb_pyspark/duckdb_pyspark_type_handler.py [hunk 1]
```diff
@@ -139,3 +145,64 @@ def my_table_a(my_table: pyspark.sql.DataFrame) -> pyspark.sql.DataFrame:
             ...
 
 """
+
+
+class DuckDBPySparkIOManager(DuckDBIOManager):
+    """An IO manager definition that reads inputs from and writes PySpark DataFrames to DuckDB. When
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-duckdb-pyspark/dagster_duckdb_pyspark/duckdb_pyspark_type_handler.py:151]:
```suggestion
    """An I/O manager definition that reads inputs from and writes PySpark DataFrames to DuckDB. When
```

File: python_modules/libraries/dagster-duckdb-pyspark/dagster_duckdb_pyspark/duckdb_pyspark_type_handler.py [hunk 2]
```diff
@@ -139,3 +145,64 @@ def my_table_a(my_table: pyspark.sql.DataFrame) -> pyspark.sql.DataFrame:
...
+            def my_table() -> pyspark.sql.DataFrame:  # the name of the asset will be the table name
+                ...
+
+            defs = Definitions(
+                assets=[my_table],
+                resources={"io_manager": DuckDBPySparkIOManager(database="my_db.duckdb")}
+            )
+
+        If you do not provide a schema, Dagster will determine a schema based on the assets and ops using
+        the IO Manager. For assets, the schema will be determined from the asset key, as in the above example.
```
Comment (erinkcochran87) [inline python_modules/libraries/dagster-duckdb-pyspark/dagster_duckdb_pyspark/duckdb_pyspark_type_handler.py:175]:
```suggestion
        the I/O Manager. For assets, the schema will be determined from the asset key, as in the above example.
```

**Review by erinkcochran87 — COMMENTED** [review]
There should probably be a technical reviewer on here as well, but the API docs look okay to me so far

**Review by benpankow — APPROVED** [review]
Looks good 👍

**Review by schrockn — APPROVED** [review]
image

**Pull request merged**